packages/postgres-database/src/simcore_postgres_database/utils_users.py

```diff
@@ -17,7 +24,57 @@ class UserNotFoundInRepoError(BaseUserRepoError):
     pass
 
 
+def _generate_username_from_email(email: str) -> str:
+    username = email.split("@")[0]
+
+    # Remove any non-alphanumeric characters and convert to lowercase
+    return re.sub(r"[^a-zA-Z0-9]", "", username).lower()
+
+
+def _generate_random_suffix() -> str:
+    return f"_{''.join(secrets.choice(string.digits) for _ in range(4))}"
+
+
 class UsersRepo:
+    @staticmethod
+    async def new_user(
+        conn: SAConnection,
+        email: str,
+        password_hash: str,
+        status: str,
+        expires_at: datetime | None,
+    ) -> RowProxy:
+        data = {
+            "name": _generate_username_from_email(email),
+            "email": email,
+            "password_hash": password_hash,
+            "status": status,
+            "role": UserRole.USER,
+            "expires_at": expires_at,
+        }
+        try:
+            user_id = await conn.scalar(
+                users.insert().values(data).returning(users.c.id)
+            )
+        except UniqueViolation:
+            data["name"] += _generate_random_suffix()
+            user_id = await conn.scalar(
+                users.insert().values(data).returning(users.c.id)
+            )
```

Comment:
NOTE: if you are unlucky, this can still raise. Say you generate the same suffix for two people with the same name.
To be on the safe side, I would retry this a few times.

Reply:
I changed it.